RecoParticleFlow/PFClusterProducer/plugins/PFClusterFromHGCalTrackster.cc

```diff
@@ -1,42 +1,42 @@
-#include "PFClusterFromHGCalMultiCluster.h"
+#include "PFClusterFromHGCalTrackster.h"
 #include "FWCore/MessageLogger/interface/MessageLogger.h"
 
 #include "FWCore/Framework/interface/Event.h"
 
 #include "DataFormats/ForwardDetId/interface/HGCalDetId.h"
 
-void PFClusterFromHGCalMultiCluster::updateEvent(const edm::Event& ev) {
-  ev.getByToken(clusterToken_, clusterH_);
+void PFClusterFromHGCalTrackster::updateEvent(const edm::Event& ev) {
   ev.getByToken(tracksterToken_, trackstersH_);
+  ev.getByToken(clusterToken_, clusterH_);
 }
 
-void PFClusterFromHGCalMultiCluster::buildClusters(const edm::Handle<reco::PFRecHitCollection>& input,
+void PFClusterFromHGCalTrackster::buildClusters(const edm::Handle<reco::PFRecHitCollection>& input,
                                                    const std::vector<bool>& rechitMask,
                                                    const std::vector<bool>& seedable,
                                                    reco::PFClusterCollection& output) {
-  const auto& hgcalMultiClusters = *clusterH_;
   auto const& hits = *input;
 
   const auto& tracksters = *trackstersH_;
+  const auto& clusters = *clusterH_;
 
   // for quick indexing back to hit energy
   std::unordered_map<uint32_t, size_t> detIdToIndex(hits.size());
   for (uint32_t i = 0; i < hits.size(); ++i) {
     detIdToIndex[hits[i].detId()] = i;
   }
 
-  int iMultiClus = -1;
-  for (const auto& mcl : hgcalMultiClusters) {
-    iMultiClus++;
-    // Skip empty multiclusters
-    if (mcl.hitsAndFractions().empty()) {
+  int iTrackster = -1;
+  for (const auto& tst : tracksters) {
+    iTrackster++;
+    // Skip empty tracksters
+    if (tst.vertices().empty()) {
       continue;
     }
     // Filter using trackster PID
     if (filterByTracksterPID_) {
       float probTotal = 0.0f;
       for (int cat : filter_on_categories_) {
-        probTotal += tracksters[iMultiClus].id_probabilities(cat);
+        probTotal += tracksters[iTrackster].id_probabilities(cat);
```

Comment:
Why not using `tst` directly? Possibly you could even drop `iTrackster`.

Reply:
Right, done.